### PR TITLE
feat(library): shared components in own repo

### PR DESCRIPTION
**Pull request by oyo**
## Description

The Catena-X portal shared components have been moved out of the portal fronted repo and development will continue in this repo. All dependencies have been upgraded to the latest versions.

## Why

The monorepo approach based on yarn workspaces caused a lot of issues with the upgrade of dependencies. Splitting the Portal frontend and shared components will alleviate development for shared components especially for third party applications who like to contribute.

## Issue

CPLP-2182 (JIRA)

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+env:
+  IMAGE_NAMESPACE: "tractusx"
+  IMAGE_NAME: "portal-frontend"
```
Comment (evegufy) [inline .github/workflows/trivy.yml:33]: env variables can be removed, no image scan.

File: src/assets/data/TableRows.json
Comment (evegufy) [inline src/assets/data/TableRows.json:1]: @oyo do you think it's possible to anonymize this file? If we don't change this now, we have it in history of main even if we change it later...

**Review by evegufy — CHANGES_REQUESTED** [review]
License review for the following dependencies in progress:
npm/npmjs/-/c8/7.14.0
npm/npmjs/-/cjs-module-lexer/1.2.3
npm/npmjs/-/core-js-compat/3.31.0
npm/npmjs/-/core-js-pure/3.31.0
npm/npmjs/-/enhanced-resolve/5.15.0
npm/npmjs/-/es-module-lexer/1.3.0
npm/npmjs/-/eslint-plugin-n/15.7.0
npm/npmjs/-/flow-parser/0.208.1
npm/npmjs/-/glob-promise/6.0.3
npm/npmjs/-/html-entities/2.3.6
npm/npmjs/-/html-webpack-plugin/5.5.3
npm/npmjs/-/jscodeshift/0.14.0
npm/npmjs/-/node-fetch-native/1.2.0
npm/npmjs/-/ramda/0.29.0
npm/npmjs/-/react-inspector/6.0.2
npm/npmjs/-/schema-utils/3.3.0
npm/npmjs/-/schema-utils/4.2.0
npm/npmjs/-/simple-update-notifier/1.1.0
npm/npmjs/-/storybook/7.0.21
npm/npmjs/-/terser/5.18.0
npm/npmjs/-/webpack/5.87.0
npm/npmjs/-/yaml/2.3.1
npm/npmjs/@babel/code-frame/7.22.5
npm/npmjs/@babel/compat-data/7.22.5
npm/npmjs/@babel/core/7.22.5
npm/npmjs/@babel/generator/7.22.5
npm/npmjs/@babel/helper-annotate-as-pure/7.22.5
npm/npmjs/@babel/helper-compilation-targets/7.22.5
npm/npmjs/@babel/helper-create-class-features-plugin/7.22.5
npm/npmjs/@babel/helper-create-regexp-features-plugin/7.22.5
npm/npmjs/@babel/helper-environment-visitor/7.22.5
npm/npmjs/@babel/helper-function-name/7.22.5
npm/npmjs/@babel/helper-hoist-variables/7.22.5
npm/npmjs/@babel/helper-member-expression-to-functions/7.22.5
npm/npmjs/@babel/helper-module-imports/7.22.5
npm/npmjs/@babel/helper-module-transforms/7.22.5
npm/npmjs/@babel/helper-optimise-call-expression/7.22.5
npm/npmjs/@babel/helper-plugin-utils/7.22.5
npm/npmjs/@babel/helper-remap-async-to-generator/7.22.5
npm/npmjs/@babel/helper-replace-supers/7.22.5
npm/npmjs/@babel/helper-simple-access/7.22.5
npm/npmjs/@babel/helper-skip-transparent-expression-wrappers/7.22.5
npm/npmjs/@babel/helper-split-export-declaration/7.22.5
npm/npmjs/@babel/helper-string-parser/7.22.5
npm/npmjs/@babel/helper-validator-identifier/7.22.5
npm/npmjs/@babel/helper-validator-option/7.22.5
npm/npmjs/@babel/helper-wrap-function/7.22.5
npm/npmjs/@babel/helpers/7.22.5
npm/npmjs/@babel/highlight/7.22.5
npm/npmjs/@babel/plugin-bugfix-safari-id-destructuring-collision-in-function-expression/7.22.5
npm/npmjs/@babel/plugin-syntax-flow/7.22.5
npm/npmjs/@babel/plugin-syntax-import-assertions/7.22.5
npm/npmjs/@babel/plugin-syntax-jsx/7.22.5
npm/npmjs/@babel/plugin-syntax-typescript/7.22.5
npm/npmjs/@babel/plugin-transform-arrow-functions/7.22.5
npm/npmjs/@babel/plugin-transform-async-to-generator/7.22.5
npm/npmjs/@babel/plugin-transform-block-scoped-functions/7.22.5
npm/npmjs/@babel/plugin-transform-block-scoping/7.22.5
npm/npmjs/@babel/plugin-transform-classes/7.22.5
npm/npmjs/@babel/plugin-transform-computed-properties/7.22.5
npm/npmjs/@babel/plugin-transform-destructuring/7.22.5
npm/npmjs/@babel/plugin-transform-dotall-regex/7.22.5
npm/npmjs/@babel/plugin-transform-duplicate-keys/7.22.5
npm/npmjs/@babel/plugin-transform-dynamic-import/7.22.5
npm/npmjs/@babel/plugin-transform-exponentiation-operator/7.22.5
npm/npmjs/@babel/plugin-transform-flow-strip-types/7.22.5
npm/npmjs/@babel/plugin-transform-for-of/7.22.5
npm/npmjs/@babel/plugin-transform-function-name/7.22.5
npm/npmjs/@babel/plugin-transform-literals/7.22.5
npm/npmjs/@babel/plugin-transform-member-expression-literals/7.22.5
npm/npmjs/@babel/plugin-transform-modules-amd/7.22.5
npm/npmjs/@babel/plugin-transform-modules-commonjs/7.22.5
npm/npmjs/@babel/plugin-transform-modules-umd/7.22.5
npm/npmjs/@babel/plugin-transform-object-super/7.22.5
npm/npmjs/@babel/plugin-transform-property-literals/7.22.5
npm/npmjs/@babel/plugin-transform-react-display-name/7.22.5
npm/npmjs/@babel/plugin-transform-react-jsx-development/7.22.5
npm/npmjs/@babel/plugin-transform-react-jsx/7.22.5
npm/npmjs/@babel/plugin-transform-react-pure-annotations/7.22.5
npm/npmjs/@babel/plugin-transform-regenerator/7.22.5
npm/npmjs/@babel/plugin-transform-reserved-words/7.22.5
npm/npmjs/@babel/plugin-transform-shorthand-properties/7.22.5
npm/npmjs/@babel/plugin-transform-spread/7.22.5
npm/npmjs/@babel/plugin-transform-sticky-regex/7.22.5
npm/npmjs/@babel/plugin-transform-template-literals/7.22.5
npm/npmjs/@babel/plugin-transform-typeof-symbol/7.22.5
npm/npmjs/@babel/plugin-transform-typescript/7.22.5
npm/npmjs/@babel/plugin-transform-unicode-escapes/7.22.5
npm/npmjs/@babel/plugin-transform-unicode-regex/7.22.5
npm/npmjs/@babel/preset-flow/7.22.5
npm/npmjs/@babel/preset-react/7.22.5
npm/npmjs/@babel/preset-typescript/7.22.5
npm/npmjs/@babel/register/7.22.5
npm/npmjs/@babel/template/7.22.5
npm/npmjs/@babel/traverse/7.22.5
npm/npmjs/@babel/types/7.22.5
npm/npmjs/@emotion/react/11.11.1
npm/npmjs/@mui/core-downloads-tracker/5.13.4
npm/npmjs/@mui/material/5.13.5
npm/npmjs/@mui/private-theming/5.13.1
npm/npmjs/@mui/styled-engine/5.13.2
npm/npmjs/@mui/system/5.13.5
npm/npmjs/@mui/utils/5.13.1
npm/npmjs/@ndelangen/get-tarball/3.0.9
npm/npmjs/@rollup/plugin-commonjs/25.0.1
npm/npmjs/@rollup/plugin-node-resolve/15.1.0
npm/npmjs/@storybook/addon-actions/7.0.21
npm/npmjs/@storybook/addon-backgrounds/7.0.21
npm/npmjs/@storybook/addon-controls/7.0.21
npm/npmjs/@storybook/addon-docs/7.0.21
npm/npmjs/@storybook/addon-essentials/7.0.21
npm/npmjs/@storybook/addon-highlight/7.0.21
npm/npmjs/@storybook/addon-links/7.0.21
npm/npmjs/@storybook/addon-measure/7.0.21
npm/npmjs/@storybook/addon-outline/7.0.21
npm/npmjs/@storybook/addon-toolbars/7.0.21
npm/npmjs/@storybook/addon-viewport/7.0.21
npm/npmjs/@storybook/addons/7.0.21
npm/npmjs/@storybook/api/7.0.21
npm/npmjs/@storybook/blocks/7.0.21
npm/npmjs/@storybook/builder-manager/7.0.21
npm/npmjs/@storybook/builder-webpack5/7.0.21
npm/npmjs/@storybook/channel-postmessage/7.0.21
npm/npmjs/@storybook/channel-websocket/7.0.21
npm/npmjs/@storybook/channels/7.0.21
npm/npmjs/@storybook/cli/7.0.21
npm/npmjs/@storybook/client-api/7.0.21
npm/npmjs/@storybook/client-logger/7.0.21
npm/npmjs/@storybook/codemod/7.0.21
npm/npmjs/@storybook/components/7.0.21
npm/npmjs/@storybook/core-client/7.0.21
npm/npmjs/@storybook/core-common/7.0.21
npm/npmjs/@storybook/core-events/7.0.21
npm/npmjs/@storybook/core-server/7.0.21
npm/npmjs/@storybook/core-webpack/7.0.21
npm/npmjs/@storybook/csf-plugin/7.0.21
npm/npmjs/@storybook/csf-tools/7.0.21
npm/npmjs/@storybook/csf/0.1.1
npm/npmjs/@storybook/docs-tools/7.0.21
npm/npmjs/@storybook/manager-api/7.0.21
npm/npmjs/@storybook/manager/7.0.21
npm/npmjs/@storybook/node-logger/7.0.21
npm/npmjs/@storybook/postinstall/7.0.21
npm/npmjs/@storybook/preset-react-webpack/7.0.21
npm/npmjs/@storybook/preview-api/7.0.21
npm/npmjs/@storybook/preview/7.0.21
npm/npmjs/@storybook/react-dom-shim/7.0.21
npm/npmjs/@storybook/react-webpack5/7.0.21
npm/npmjs/@storybook/react/7.0.21
npm/npmjs/@storybook/router/7.0.21
npm/npmjs/@storybook/store/7.0.21
npm/npmjs/@storybook/telemetry/7.0.21
npm/npmjs/@storybook/theming/7.0.21
npm/npmjs/@storybook/types/7.0.21
npm/npmjs/@testing-library/dom/9.3.1
npm/npmjs/@testing-library/react/14.0.0
npm/npmjs/@types/babel__traverse/7.20.1
npm/npmjs/@types/detect-port/1.3.3
npm/npmjs/@types/prettier/2.7.3

File: src/assets/data/TableRows.json
Comment (oyo) [reply src/assets/data/TableRows.json:1]: good idea - done

**Comment by evegufy** [comment]
> License review for the following dependencies in progress: ...

All license review requests have been approved by dash bot user, with the exception of https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8976 but I assume this one will be approved as soon as a person from EF support looks at it as the previous version has been approved as well https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/3237#note_1013840

**Review by evegufy — APPROVED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**